src/RustyIceberg.jl

```diff
@@ -64,297 +30,600 @@ else
 end
 
 """
-    load_iceberg_library(lib_path::String=iceberg_rust_ffi_jll.libiceberg_rust_ffi)
+Runtime configuration for the Iceberg library.
+"""
+struct StaticConfig
+    n_threads::Culonglong
+end
+
+function default_panic_hook()
+    println("Rust thread panicked, exiting the process")
+    exit(1)
+end
+
+const _ICEBERG_STARTED = Atomic{Bool}(false)
+
+function iceberg_started()
+    return _ICEBERG_STARTED[]
+end
+
+const _INIT_LOCK::ReentrantLock = ReentrantLock()
+_PANIC_HOOK::Function = default_panic_hook
+
+Base.@ccallable function panic_hook_wrapper_iceberg()::Cint
```

Comment:
nit: it came to my mind that perhaps `iceberg` should be a prefix instead of a suffix